mono/metadata/domain-internals.h

```diff
@@ -235,11 +235,16 @@ struct _MonoJitInfo {
 
 #define MONO_SIZEOF_JIT_INFO (offsetof (struct _MonoJitInfo, clauses))
 
+typedef struct {
+	gpointer *static_data;
+	uint32_t gc_handle;
+} ContextStaticData;
+
 struct _MonoAppContext {
 	MonoObject obj;
 	gint32 domain_id;
 	gint32 context_id;
-	gpointer *static_data;
+	ContextStaticData *data;
```

Comment:
> There's no need to change this field to point to ContextStaticData instead of static_stata.

How would we access a field on `MonoAppContext` from a reference queue callback, especially when the GC handle that *possibly* keeps it alive is a weak handle?

> In this patch, there's no usage of gc_handle through the context.

We free the handle in the reference queue callback - same issue here. How would we access a field on `MonoAppContext`?